src/main/java/com/mraof/minestuck/blockentity/machine/GristCollectorBlockEntity.java

```diff
@@ -0,0 +1,88 @@
+package com.mraof.minestuck.blockentity.machine;
+
+import com.mraof.minestuck.alchemy.*;
+import com.mraof.minestuck.block.machine.GristCollectorBlock;
+import com.mraof.minestuck.blockentity.MSBlockEntityTypes;
+import com.mraof.minestuck.entity.item.GristEntity;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+
+import java.util.List;
+
+public class GristCollectorBlockEntity extends BlockEntity
+{
+	private GristSet storedGrist = new GristSet();
+	
+	public GristCollectorBlockEntity(BlockPos pos, BlockState state)
+	{
+		super(MSBlockEntityTypes.GRIST_COLLECTOR.get(), pos, state);
+	}
+	
+	public GristSet getStoredGrist()
+	{
+		return storedGrist;
```

Comment:
```suggestion
		return storedGrist.asImmutable();
```

Ensures that the grist set is not modified outside this class by modifying the returned grist set.